pilot/pkg/security/trustdomain/bundle.go

```diff
@@ -115,17 +134,17 @@ func getTrustDomain(principal string) string {
 	// A valid SPIFFE identity in authorization has no SPIFFE:// prefix.
 	// It is presented as <trust-domain>/ns/<some-namespace>/sa/<some-service-account>
 	if len(identityParts) != 5 {
-		rbacLog.Errorf("Wrong SPIFFE format: %s", principal)
 		return ""
 	}
 	return identityParts[0]
 }
 
-func found(key string, list []string) bool {
-	for _, l := range list {
-		if key == l {
-			return true
-		}
+// getTrustDomainOrError is similar to getTrustDomain. However, it expects principal to have the right
+// SPIFFE format and the trust domain cannot be * (i.g. */ns/foo/sa/bar).
```

Comment:
Why trust domain cannot be *? "*/ns/foo/sa/bar" looks valid.

Reply:
For the given principal, the trust domain is "", and we already check for that case here https://github.com/istio/istio/blob/21ddfc6d4744b0a08dbd2fa628cf3c30ce5904d7/pilot/pkg/security/trustdomain/bundle.go#L65.

This function `getTrustDomainOrError` simply finds trust domain that are not "*" or "" (these trust domains mean they will match to any trust domain).

Reply:
Actually, "/ns/foo/sa/bar" is different from "*/ns/foo/sa/bar". The first one is exact match, the second one is suffix match. The current PR doesn't handle this. I'll push some code for this.

Reply:
Pushed. PTAL.

Reply:
Actually I was asking about "\*", and "\*/ns/foo/sa/bar". I forgot to put the escape character.